parlai/tasks/dailydialog/agents.py

```diff
@@ -88,3 +88,44 @@ def share(self):
         shared = super().share()
         shared['data'] = self.data
         return shared
+
+
+class NoStartTeacher(Convai2Teacher):
+    """ Same as default teacher, but it doesn't contain __SILENCE__ entries.
+        If we are the first speaker, then the first utterance is skipped.
+    """
+    def __init__(self, opt, shared=None):
+        super().__init__(opt, shared)
+
+        # Calculate the correct number of examples.
+        self.num_exs = sum(len(d['dialogue']) - 1 for d in self.data)
+
+        # Store all episodes in self.data
+        self.all_eps = self.data + [d for d in self.data if len(d['dialogue']) > 2]
+        self.num_eps = len(self.all_eps)
+
+    def get(self, episode_idx, entry_idx=0):
+        full_eps = self.all_eps[episode_idx]
+        entries = full_eps['dialogue']
+
+        # Sometimes we're speaker 1 and sometimes we're speaker 2.
+        # We can't be speaker 1 if dialog has only 2 turns.
+        speaker_id = episode_idx % 2 if len(entries) > 2 else 0
```

Comment:
I think this should now be `speaker_id = int(episode_idx >= len(self.data))`, right?

Reply:
Yes, that's actually a lot better.